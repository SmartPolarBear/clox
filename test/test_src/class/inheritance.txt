R"(
/*class Doughnut {
  fun cook() {
    print "Fry until golden brown.";
  }
}

var dn=Doughnut();
dn.cook();


class BostonCream : Doughnut {}

BostonCream().cook();*/

class A {
  fun method() {
    print "A method";
  }
}

class B : A {
  fun method() {
    print "B method";
  }

  fun test() {
    this.method();
    base.method();
  }
}

class C : B {}

C().test();

)"