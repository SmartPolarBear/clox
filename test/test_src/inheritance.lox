class Doughnut {
  cook() {
    print "Fry until golden brown.";
  }
}

class BostonCream : Doughnut {}

BostonCream().cook();

class A {
  method() {
    print "A method";
  }
}

class B : A {
  method() {
    print "B method";
  }

  test() {
    base.method();
  }
}

class C : B {}

C().test();