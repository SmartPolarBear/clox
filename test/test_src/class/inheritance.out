R"(B method
A method)"